Compiler error (GCC 14.2.0 on Linux)

### Check duplicate issues.

- [X] Checked for duplicates

### Description

Compilation fails

### Reproducer

Following the instructions from [https://root.cern/install/build_from_source/](https://root.cern/install/build_from_source/):
```
git clone --branch latest-stable --depth=1 https://github.com/root-project/root.git src
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=$PWD/../install ../src
make -j 24
```
on a Rocky Linux 8.10 system with GCC 14.2.0, GLIBC 2.28, Clang 18.1.8 gives a compiler error 
```
[ 62%] Building CXX object interpreter/llvm-project/llvm/tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CGObjCRuntime.cpp.o
CMake Error at /home/aehle/g/build/builtins/davix/DAVIX-prefix/src/DAVIX-stamp/DAVIX-build-.cmake:49 (message):
  Command failed: 2

   'make'

  See also

    /home/aehle/g/build/builtins/davix/DAVIX-prefix/src/DAVIX-stamp/DAVIX-build-*.log
```
The first error in the log referenced here is 
```
/home/aehle/g/build/builtins/davix/DAVIX-prefix/src/DAVIX/src/libs/rapidjson/document.h: In member function ‘rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)’:
/home/aehle/g/build/builtins/davix/DAVIX-prefix/src/DAVIX/src/libs/rapidjson/document.h:319:82: error: assignment of read-only member ‘rapidjson::GenericStringRef<CharType>::length’
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
```
Logs:
[cmake.log](https://github.com/user-attachments/files/18005948/cmake.log)
[make.log](https://github.com/user-attachments/files/18005951/make.log)
[DAVIX-build-err.log](https://github.com/user-attachments/files/18005949/DAVIX-build-err.log)
[DAVIX-build-out.log](https://github.com/user-attachments/files/18005950/DAVIX-build-out.log)



### ROOT version

commit 1b923a47801d73194d763212d2290b878f3b8908 (current latest-stable)

### Installation method

build from source

### Operating system

Rocky linux 8.10 (Green Obsidian)

### Additional context

Seems to be related to an old rapidjson bug https://github.com/Tencent/rapidjson/issues/2277 that was fixed in 2016 but there has not been a rapidjson release since. ROOT's version of Davix comes with a copy of rapidjson that contains this bug (reported in https://github.com/cern-fts/davix/issues/130).

--- a/builtins/davix/CMakeLists.txt
+++ b/builtins/davix/CMakeLists.txt
@@ -10,9 +10,9 @@ find_package(libuuid REQUIRED)
 find_package(LibXml2 REQUIRED)
 find_package(OpenSSL REQUIRED)
 
-set(DAVIX_VERSION "0.8.7")
+set(DAVIX_VERSION "0.8.7p1")
 set(DAVIX_URL "http://lcgpackages.web.cern.ch/lcgpackages/tarFiles/sources")
-set(DAVIX_URLHASH "SHA256=78c24e14edd7e4e560392d67147ec8658c2aa0d3640415bdf6bc513afcf695e6")
+set(DAVIX_URLHASH "SHA256=d4eee9f20aa032893ce488273cc0bfb62bcad8e2a1afa6b260130508eaf3ce54")
 set(DAVIX_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/DAVIX-prefix)
 set(DAVIX_LIBNAME ${CMAKE_STATIC_LIBRARY_PREFIX}davix${CMAKE_STATIC_LIBRARY_SUFFIX})
 

